suite-native/module-passphrase/src/passphraseThunks.ts

```diff
@@ -0,0 +1,31 @@
+import { createThunk } from '@suite-common/redux-utils';
+import {
+    deviceActions,
+    selectDevice,
+    selectDeviceThunk,
+    selectDevices,
+} from '@suite-common/wallet-core';
+import TrezorConnect from '@trezor/connect';
+
+const PASSPHRASE_MODULE_PREFIX = '@suite-native/device';
+
+export const cancelPassphraseAndSelectPreviousDeviceThunk = createThunk(
+    `${PASSPHRASE_MODULE_PREFIX}/cancelPassphraseFlow`,
+    (_, { getState, dispatch }) => {
+        const devices = selectDevices(getState());
+        const device = selectDevice(getState());
+
+        if (!device) return;
+
+        // Select standard wallet (e.g. empty passphrase) that has the same device ID.
+        const standardWalletDeviceIndex = devices.findIndex(
```

Comment:
I believe this is more bullet proof than selecting based on `useEmptyPassphrase` true. When we will have multiple connected devices, this condition would fail. Matching by `device.id` and `instance` seems better to me.